### PR TITLE
[IOPLT-624] Infrastructure refactoring

### PR DESCRIPTION
**Pull request by Krusty93**
#### List of Changes

Function App renaming
Remove reference to IO private DNS zone for `azurewebsites`
Switch to DX TF modules

#### Motivation and Context

Refactoring before going live

#### Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] I have committed only changes relevant to the task.
- [ ] I have minimized the number of changes.
- [ ] My changes are about only one task or issue.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/modules/commons/variables.tf [hunk 1]
```diff
@@ -47,13 +47,13 @@ variable "snet_pendpoints_address_space" {
 }
 
 
-variable "cidr_subnet_fnsubscriptionasync" {
-  type        = list(string)
+variable "cidr_subnet_func_consumptions" {
```
Comment (kin0992) [inline infra/resources/modules/commons/variables.tf:50]: **suggestion**: should we follow the rename?

```suggestion
variable "cidr_subnet_func_consumers" {
```

Comment (kin0992) [reply infra/resources/modules/commons/variables.tf:50]: **nitpick**: just to keep it clear, I'd update the description as well

File: infra/resources/modules/commons/variables.tf [hunk 2]
```diff
@@ -47,13 +47,13 @@ variable "snet_pendpoints_address_space" {
...
+  type        = string
   description = "The subscription async function address space"
 }
 
-variable "cidr_subnet_fnsubscription" {
-  type        = list(string)
+variable "cidr_subnet_func_api" {
+  type        = string
   description = "The subscription function address space"
```
Comment (kin0992) [inline infra/resources/modules/commons/variables.tf:57]: **nitpick**: same here, I'd update the description

File: infra/resources/modules/commons/function_consumptions.tf [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    name                = azurerm_virtual_network.vnet.name
+    resource_group_name = azurerm_resource_group.net_rg.name
+  }
+  private_dns_zone_resource_group_name = azurerm_resource_group.net_rg.name
+
+  tags = var.tags
+}
+
+# Enables the consumers_fn to write on service-bus topic
+resource "azurerm_role_assignment" "subs_asyn_write_to_sbt" {
```
Comment (gunzip) [inline infra/resources/modules/commons/function_consumptions.tf:99]: can't we refactor this part using the new role module? (I mean, in another PR)

File: infra/resources/modules/commons/function_consumptions.tf [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+  scheduler = {
+    normal_load = {
+      default = 3
+      minimum = 3
+    }
+    maximum = 30
+  }
+
+  scale_metrics = {
```
Comment (gunzip) [inline infra/resources/modules/commons/function_consumptions.tf:245]: we need a preset for this @kin0992

**Review by datalek — APPROVED** [review]

**Pull request merged**